Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -89,6 +89,61 @@ public class GraphLinter: GraphLinting {
 
     private func lintDependencies(graphTraverser: GraphTraversing, config: Config) -> [LintingIssue] {
         var issues: [LintingIssue] = []
+
+        issues.append(contentsOf: lintDependencyRelationships(graphTraverser: graphTraverser))
+        issues.append(contentsOf: lintLinkableDependencies(graphTraverser: graphTraverser))
+        issues.append(contentsOf: staticProductsLinter.lint(graphTraverser: graphTraverser, config: config))
+        issues.append(contentsOf: lintPrecompiledFrameworkDependencies(graphTraverser: graphTraverser))
+        issues.append(contentsOf: lintPackageDependencies(graphTraverser: graphTraverser))
+        issues.append(contentsOf: lintAppClip(graphTraverser: graphTraverser))
+
+        return issues
+    }
+
+    private func lintLinkableDependencies(graphTraverser: GraphTraversing) -> [LintingIssue] {
+        let linkableProducts: Set<Product> = [
+            .framework,
+            .staticFramework,
+            .staticLibrary,
+            .dynamicLibrary,
+        ]
+
+        let dependencyIssues = graphTraverser.dependencies.flatMap { fromDependency, _ -> [LintingIssue] in
+            guard case let GraphDependency.target(fromTargetName, fromTargetPath) = fromDependency,
+                  let fromTarget = graphTraverser.target(path: fromTargetPath, name: fromTargetName) else { return [] }
+
+            let fromPlatforms = fromTarget.target.supportedPlatforms
+
+            let dependencies: [LintingIssue] = graphTraverser.directTargetDependencies(path: fromTargetPath, name: fromTargetName)
+                .flatMap { dependentTarget in
+                    guard linkableProducts.contains(dependentTarget.target.product) else { return [LintingIssue]() }
+
+                    var requiredPlatforms = fromPlatforms
+
+                    if let condition = dependentTarget.condition {
+                        requiredPlatforms.formIntersection(Set(condition.platformFilters.compactMap(\.platform)))
+                    }
+
+                    let unaccountedPlatforms = requiredPlatforms.subtracting(dependentTarget.target.supportedPlatforms)
+
+                    if !unaccountedPlatforms.isEmpty {
+                        let missingPlatforms = unaccountedPlatforms.map(\.rawValue).joined(separator: ", ")
+                        return [LintingIssue(
+                            reason: "Target \(fromTargetName) with depends on \(dependentTarget.target.name) and but does not support the required platforms \(missingPlatforms). This dependency requires a condition.",
```

Comment:
What do you think of making this message a little more verbose ? 
```suggestion
                            reason: "Target \(fromTargetName) which \(dependentTarget.target.name) is depending on, does not support the required platforms \(missingPlatforms). \(fromTargetName) requires a condition which restricts it to at most \(missingPlatforms)",
```

Reply:
i like it!